cli/azd/cmd/testdata/TestUsage-azd-provision.snap

```diff
@@ -13,6 +13,7 @@ Flags
         --docs               	: Opens the documentation for azd provision in your web browser.
     -e, --environment string 	: The name of the environment to use.
     -h, --help               	: Gets help for provision.
+        --what-if            	: Preview changes to Azure resources.
```

Comment:
`--what-if` feels bicep centric given that we will also adopt TF's plan eventually. 

Any other options like `preview` `dry-run` etc?

@ellismg @savannahostrowski @vhvb1989

Reply:
I was also thinking about this. I'd probably lean toward `--preview` over `--dry-run`

Reply:
I like `--preview` as well and it matches the copy that @vhvb1989 used in the user facing messages (i.e. "Preview Azure resource changes").